### DISCUSSION TITLE
Ordering relationship dropdown?

### DISCUSSION THREAD

**Discussion by Grayda** [Budibase/budibase#4579] [Help] [answered]
I've got two tables in my app, `Items` and `Platforms`. In the `Items` table, the user can pick a Platform as a relationship. Some platforms are more popular than others, so in the `Platforms` table I've got a boolean column, `Favourites`. I'd like to be able to sort the relationship dropdown so the favourites appear at the top of the list for quick picking, followed by the remaining items. Right now they're just sorted alphabetically.

Is it possible to sort a relationship dropdown box by a field? If not, is this an easy feature to add?

**Comment by mjashanks** [comment, marked as answer]
Hey David. Good shout, we could do with a few more options on the relationship picker.

You can set this up using an options picker though...

1. Remove your relationship picker
2. Add a `Data Provider`, linked to your `Platforms` table, ordered by `Favourites`
3. Inside the data provider, add an `Options Picker`
4. For the picker's `Field`, manually type `Platform` - i.e. the column name of the relationship
5. Set the picker's `Schema` to Data Provider... and configure the options below that.

That should be it - I just tried it and it works for me!

**Reply by agaheu** [reply]
Hi,
Yes, it works well while creating new row. However, in update screen although the "Option Picker" shows all options correctly there is no selected option and below is seen ([object Object]) :

![image](https://user-images.githubusercontent.com/3132816/156904767-53f4e36e-0ce3-4181-b59c-67413f1a2cfa.png)

**Reply by mjashanks** [reply]
Yea, I see what you mean.

In this case you need to 
1. use the `Default Value` field, to pick the currently set value: e.g. `{{ Repeater.Items.Platform.0._id }}`
2. Set your select's field name to something other than the column name - e.g. `Platform_picker`
3. In your `Save Row` action, manually set the `Platform` field to `{{ Form.Fields.Platform_picker }}` (i.e. the value of the select)

**Reply by agaheu** [reply]
it works :) thanks...

**Reply by juangbm** [reply]
Hi @mjashanks !

I was testing it, but I am using a Many relationship and this is not working, I can display the different options filtered from the data provider I have added, but when I select them and try to save, I get an error "row[key].map is not a function".

Any idea?

Thanks!

**Reply by Jednadvacet** [reply]
Hello, just a question: but how to save the value from new (filtered) option field to database column that is relationship type. 
Is it possible somehow?

Thanks for reply in advance.
Regards, 
Ondrej

**Comment by melohagan** [comment]
Hey @Jednadvacet 

If you're saving relationships for a SQL table, then you would just save the values to the foreign keys as necessary, and the Budibase relationship field will automatically reflect the link.

When using relationships, we recommend SQL.

For a Budibase DB, you'll want to save an **_id** against the relationship field. 

So it is critical that the **_id** field is the *Value* field of your options picker:

![Screenshot 2023-08-04 at 16 40 14](https://github.com/Budibase/budibase/assets/101575380/ea7c8ac4-68c8-49c4-a2c1-efa4687d35c4)

Then in your save button, you'll need to `Add Column`, select the relationship field and use the options picker binding.
 
![Screenshot 2023-08-04 at 16 44 23](https://github.com/Budibase/budibase/assets/101575380/304b172e-a927-4c76-a352-17afd4fc8d72)
